blocks/rich-text/index.js

```diff
@@ -692,12 +667,22 @@ export class RichText extends Component {
 		} );
 	}
 
-	setContent( content = '' ) {
-		this.editor.setContent( renderToString( content ) );
+	setContent( content ) {
+		const { format } = this.props;
+		this.editor.setContent( valueToString( content, format ) );
 	}
 
 	getContent() {
-		return nodeListToReact( this.editor.getBody().childNodes || [], createTinyMCEElement );
+		const { format } = this.props;
+
+		switch ( format ) {
+			case 'string':
+				return this.editor.getContent();
+			default:
+				return this.editor.dom.isEmpty( this.editor.getBody() ) ?
+					[] :
+					domToFormat( this.editor.getBody().childNodes || [], 'element' );
```

Comment:
Aside: In the future, I might imagine we avoid working with the DOM as a source and instead defer to TinyMCE's `getContent( { format: 'tree' } )` to convert from their tree form to the desired target format. This would avoid the need for us to maintain our own cleaning logic. At which point, `domToFormat` doesn't make as much sense, might just be `toFormat( value, targetFormat, sourceFormat )` (maybe infer source format?).

Reply:
I tend to always prefer explicitness over implicitness. I don't really care about the name of the function but I tend to be suspicious (maybe I shoudn't) every time I hear "infer from value"

Reply:
Eh, I can see it, but it's kinda like if one had to pass a second argument to e.g. `Array.from`:

```js
Array.from( new Set( [ 1, 2, 3 ] ), 'Set' );
```

... one could say: "Well, duh, you passed me a set, I know it's a set"

Reply:
But sometimes it's not obvious, the easy example is that the tree structure and the element structure can't be distinguished easily because both can be plain objects.